frontend/util.lua

```diff
@@ -210,6 +210,52 @@ function util.tableSize(t)
     return count
 end
 
+--- Returns a value of a key, checks if all parent keys are not empty.
+---- @param t Lua table
+---- @param ... parent keys, starting from the upper level
+---- @treturn value of the last key or nil
+function util.tableGetValue(t, ...)
+    local keys = { ... }
+    local q = t
+    for _, key in ipairs(keys) do
+        if type(q) ~= "table" then return end
+        q = q[key]
+        if q == nil then return end
+    end
+    return q
+end
+
+--- Sets a value of a key, creates all parent keys if needed.
+---- @param t Lua table
+---- @param value value to be assigned to the last key
+---- @param ... parent keys, starting from the upper level
+function util.tableSetValue(t, value, ...)
+    local keys = { ... }
+    local q = t
+    for i = 1, #keys - 1 do
+        local key = keys[i]
+        q[key] = q[key] or {}
+        q = q[key]
+    end
+    q[keys[#keys]] = value
+end
+
+--- Removes a key in a table, removes all empty parent keys.
+---- @param t Lua table
+---- @param ... parent keys, starting from the upper level
+function util.tableRemovePath(t, ...)
+    local keys = { ... }
```

Comment:
Note sure "Path" is the right word for some generic table function - it's too much "file system" connoted.
May be tableRemoveSubTree ?

Reply:
The purpose is to delete a value of the (last) key. Removing empty subtree is a consequence. So maybe `util.tableRemoveValue` or `util.tableDelValue` (like Get and Set).

Reply:
> So maybe `util.tableRemoveValue` or `util.tableDelValue` (like Get and Set).

Fine with me (rather full words like Remove or Delete, than just Del).